src/consensus.rs

```diff
@@ -138,10 +152,10 @@ impl<T: Proposition> Consensus<T> {
             let decision = Decision {
                 votes,
                 proposals,
-                faults: self.faults(),
+                faults: signed_vote.vote.faults.clone(),
```

Comment:
Again, similar problem, we were adopting a decision from the voter, but we were using the faults we had seen instead of the faults the sender had seen. So our decision will potentially have different proposals if we see different faults.